### PR TITLE
refactor(client): extract hard coded key codes

### LINKED ISSUE #3163: Several instances of hard coded key codes
Follow on from #3122. There are a several hard coded key codes (like `13` for the enter key). Instead of hard coding and wondering, "hey, what does 13 mean?", we should create a new file, `lib/key-codes.js` that has the key codes for the keys we check.

### PR DESCRIPTION
**Pull request by npestana**
Extract hard coded key codes to 'lib/key-codes.js'.

Closes #3163

### CODE REVIEW

File: app/scripts/lib/key-codes.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+ define(function () {
+   'use strict';
+
+   return {
+     down_arrow: 40,
```
Comment (vladikoff) [inline app/scripts/lib/key-codes.js]: should these be UPPER_CASE like other constants?

File: app/scripts/views/tooltip.js [hunk 2]
```diff
@@ -27,14 +28,6 @@ define([
...
       this.extraClassNames = options.extraClassNames || '';
 
-      // Event codes for the keyboard keys.
-      this.DOWN_ARROW_KEY_CODE = 40;
-      this.ENTER_KEY_CODE = 13;
-      this.LEFT_ARROW_KEY_CODE = 37;
-      this.RIGHT_ARROW_KEY_CODE = 39;
-      this.TAB_KEY_CODE = 9;
-      this.UP_ARROW_KEY_CODE = 38;
-
```
Comment (vladikoff) [inline app/scripts/views/tooltip.js]: related to https://github.com/mozilla/fxa-content-server/pull/3220/files#r43058886

**Comment by vladikoff** [comment]
```
/home/travis/build/mozilla/fxa-content-server/app/scripts/lib/key-codes.js
   9:6  error  Identifier 'down_arrow' is not in camel case   camelcase
  11:6  error  Identifier 'left_arrow' is not in camel case   camelcase
  12:6  error  Identifier 'right_arrow' is not in camel case  camelcase
  14:6  error  Identifier 'up_arrow' is not in camel case     camelcase
  15:5  error  Missing semicolon                              semi
✖ 5 problems (5 errors, 0 warnings)

```

**Comment by vladikoff** [comment]
@shane-tomlinson any other concerns?

File: app/scripts/lib/mailcheck.js [hunk 3]
```diff
@@ -7,9 +7,10 @@
 /* exceptsPaths: mailcheck */
 define([
   'views/tooltip',
+  'lib/key-codes',
   'mailcheck'
 ],
-function (Tooltip) {
+function (Tooltip, keyCodes) {
```
Comment (pdehaan) [inline app/scripts/lib/mailcheck.js]: Do we want to use `keyCodes` or `KeyCodes`?

File: app/scripts/lib/key-codes.js (hunk 1 above)
Comment (shane-tomlinson) [inline app/scripts/lib/key-codes.js]: :+1:

File: app/scripts/lib/mailcheck.js (hunk 3 above)
Comment (shane-tomlinson) [inline app/scripts/lib/mailcheck.js]: Let's use `KeyCodes`, like we do with `Constants`

**Comment by shane-tomlinson** [comment]
Thanks @npestana! This is a fantastic start. Can you alter the commit message to use the prefix `refactor(client):` since so far everything has to do with the client in some form?

**Comment by npestana** [comment]
@shane-tomlinson @vladikoff @pdehaan I've made the changes to reflect the coding standards and updated the pull request.

**Comment by shane-tomlinson** [comment]
I totally dig this. Running the tests now.

**Pull request merged**